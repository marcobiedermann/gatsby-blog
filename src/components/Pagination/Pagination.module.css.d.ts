/* eslint-disable camelcase */

declare const styles: {
  readonly pagination: string;
  readonly pagination__itemNext: string;
  readonly pagination__itemPrevious: string;
};
export = styles;
