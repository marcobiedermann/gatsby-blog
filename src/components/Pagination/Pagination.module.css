.pagination {
  display: grid;
  gap: 1em;
  list-style: none;
  margin-block: 0.5em 1.5em;
  padding: 0;
}

.pagination__item--next {
  justify-self: flex-end;
}

.pagination__item--previous {
  justify-self: flex-start;
}
