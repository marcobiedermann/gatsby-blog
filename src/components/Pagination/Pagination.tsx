/* eslint-disable camelcase */

import clsx from 'clsx';
import { Link } from 'gatsby';
import React from 'react';
import {
  pagination,
  pagination__itemNext,
  pagination__itemPrevious,
} from './Pagination.module.css';

export interface PaginationProps {
  next: {
    fields: {
      slug: string;
    };
  } | null;
  previous: {
    fields: {
      slug: string;
    };
  } | null;
}

function Pagination(props: PaginationProps) {
  const { next, previous } = props;

  return (
    <ul className={pagination}>
      {previous && (
        <li className={clsx(pagination__itemPrevious)}>
          <Link to={`/blog${previous.fields.slug}`}>« Previous</Link>
        </li>
      )}
      {next && (
        <li className={clsx(pagination__itemNext)}>
          <Link to={`/blog${next.fields.slug}`}>Next »</Link>
        </li>
      )}
    </ul>
  );
}

export default Pagination;
