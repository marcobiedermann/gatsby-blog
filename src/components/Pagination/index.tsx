import classNames from 'classnames';
import { Link } from 'gatsby';
import React, { FC } from 'react';
import styles from './style.module.css';

export interface PaginationProps {
  next: {
    fields: {
      slug: string;
    };
  } | null;
  previous: {
    fields: {
      slug: string;
    };
  } | null;
}

const Pagination: FC<PaginationProps> = (props) => {
  const { next, previous } = props;

  return (
    <ul className={styles.pagination}>
      {previous && (
        <li className={classNames(styles.pagination__itemPrevious)}>
          <Link to={`/blog${previous.fields.slug}`}>« Previous</Link>
        </li>
      )}
      {next && (
        <li className={classNames(styles.pagination__itemNext)}>
          <Link to={`/blog${next.fields.slug}`}>Next »</Link>
        </li>
      )}
    </ul>
  );
};

export default Pagination;
