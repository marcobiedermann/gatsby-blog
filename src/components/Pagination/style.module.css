.pagination {
  display: grid;
  gap: 1em;
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 0.5em;
  padding: 0;
}

.pagination__item--next {
  justify-self: flex-end;
}

.pagination__item--previous {
  justify-self: flex-start;
}