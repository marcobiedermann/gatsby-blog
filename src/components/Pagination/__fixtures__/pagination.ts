import { PaginationProps } from '../Pagination';

const pagination: PaginationProps = {
  next: {
    fields: {
      slug: 'next-post',
    },
  },
  previous: {
    fields: {
      slug: 'previous-post',
    },
  },
};

export default pagination;
