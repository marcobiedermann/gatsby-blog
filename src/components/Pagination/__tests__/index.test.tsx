import { render } from '@testing-library/react';
import React from 'react';
import Pagination from '..';
import pagination from '../__fixtures__';

describe('pagination component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Pagination {...pagination} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
