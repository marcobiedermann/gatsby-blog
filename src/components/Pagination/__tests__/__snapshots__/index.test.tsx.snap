// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li>
      <a
        href="/blogprevious-post"
      >
        « Previous
      </a>
    </li>
    <li>
      <a
        href="/blognext-post"
      >
        Next »
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
